nncf/api/compression.py

```diff
@@ -302,18 +303,36 @@ class CompressionAlgorithmBuilder(ABC):
     Determines which modifications should be made to the original model in
     order to enable algorithm-specific compression during fine-tuning.
     """
+    _registered_name = None  # Attribute will set by COMPRESSION_ALGORITHMS registry
 
     def __init__(self, config: NNCFConfig, should_init: bool = True):
         """
         Initializes internal state of the compression algorithm builder
 
-        :param config: The dictionary that contains parameters of the compression
-            method.
+        :param config: The top-level NNCFConfig object (i.e. parsed from a .json and extended with
+            all necessary objects required for compression such as initialization data loaders).
         :param should_init: If False, trainable parameter initialization will be
             skipped during building.
         """
         self.config = config
         self.should_init = should_init
+        self._algo_config = self._get_algo_specific_config_section()
+
+        self.ignored_scopes = self.config.get('ignored_scopes')
+        self.target_scopes = self.config.get('target_scopes')
+
+        if "ignored_scopes" in self._algo_config:
+            algo_ignored_scopes = self._algo_config["ignored_scopes"]
```

Comment:
Please use `'` instead of `"` here and elsewhere

Reply:
Done